vfsutil/zip.go

```diff
@@ -54,39 +55,50 @@ func NewZipVFS(ctx context.Context, url string, onFetchStart, onFetchFailed func
 			MaxCacheSizeBytes: MaxCacheSizeBytes,
 		}
 
-		ff, err := cachedFetch(ctx, url, store, func(ctx context.Context) (io.ReadCloser, error) {
+		// Create a new URL that doesn't include the user:password (the access
+		// token) so that the same repository at a revision for a different user
+		// results in a cache hit.
+		urlStruct, err := url.Parse(urlString)
+		if err != nil {
+			return nil, err
+		}
+		urlStruct.User = nil
+		urlWithoutUser := urlStruct.String()
+
+		ff, err := cachedFetch(ctx, urlWithoutUser, store, func(ctx context.Context) (io.ReadCloser, error) {
```

Comment:
A cleaner thing to do would just write the auth info stripping as a function:

```Go
func withoutAuthentication(url string) string
```

Then use it here:

Reply:
you are right this func signature would need to rename `url` to `u` though in this case as it would conflict with the import